### DISCUSSION TITLE
v3.100.0

### DISCUSSION THREAD

**Discussion by justinvp** [pulumi/pulumi#15040] [Releases]
### Features

- [auto/{go,nodejs,python}] Adds ListEnvironments support to Go, Nodejs and Python Automation API.
  [#14995](https://github.com/pulumi/pulumi/pull/14995)

- [cli/display] Warn correctly about new versions being availablen when using the CLI dev channel
  [#14954](https://github.com/pulumi/pulumi/pull/14954)

### Bug Fixes

- [cli/new] Fixes duplicate printing of the AI prompt & answer on Windows.
  [#15010](https://github.com/pulumi/pulumi/pull/15010)

- [engine] Fix bug where unimported resources (of different types) with the same name would cause `--import-file` to panic.
  [#15008](https://github.com/pulumi/pulumi/pull/15008)

- [engine] Default parallel operations to a reasonable limit.
  [#15016](https://github.com/pulumi/pulumi/pull/15016)

- [programgen] Fixes type-annotating nested resource properties when these have quoted keys
  [#15001](https://github.com/pulumi/pulumi/pull/15001)

- [sdkgen] Temporarily revert validation of `urn` and `id` resource output properties
  [#15025](https://github.com/pulumi/pulumi/pull/15025)

- [programgen/{dotnet,python}] Allow object keys to be template expressions such as static quoted strings
  [#15000](https://github.com/pulumi/pulumi/pull/15000)

- [backend/service] Exclude internal events from those sent to Pulumi Cloud.
  [#14972](https://github.com/pulumi/pulumi/pull/14972)

- [sdk/go] fixes fulfillValue error when joining a nil output state
  [#14949](https://github.com/pulumi/pulumi/pull/14949)

### Miscellaneous

- [pkg] Upgrade pulumi-java to v0.9.9
  [#15007](https://github.com/pulumi/pulumi/pull/15007)

- [yaml] Upgrade yaml to v1.4.5
  [#15007](https://github.com/pulumi/pulumi/pull/15007)

<hr /><em>This discussion was created from the release <a href='https://github.com/pulumi/pulumi/releases/tag/v3.100.0'>v3.100.0</a>.</em>

**Comment by sergeevabc** [comment]
`B3SUMS` here is a list of BLAKE3 checksums (hashes) to be verified, for example, by means of [RapidCRC](https://www.ov2.eu/programs/rapidcrc-unicode). But what app do I need to process `B3SUMS.sig`? Its contents do not look like the output of [Minisign](https://github.com/jedisct1/minisign) or GPG. Is it, err, [Cosign](https://docs.sigstore.dev/verifying/verify/)?

```console
$ cosign-windows-amd64.exe verify B3SUMS.sig
Error setting console mode: The parameter is incorrect.
Error: --certificate-identity or --certificate-identity-regexp is required for verification in keyless mode
main.go:74: error during command execution: --certificate-identity or --certificate-identity-regexp is required for verification in keyless mode
```
